### ISSUE TITLE
Update Request: Silicon support version

### ISSUE THREAD

**Issue by sameen-d-reza** [kmikiy/SpotMenu#257]
### Prerequisites

Please check that your issue isn't already filed: https://github.com/kmikiy/SpotMenu/issues?utf8=✓&q=

### Description

[Can you make a version which is native version instead of Intel]

### Steps to Reproduce

1. [First Step]
2. [Second Step]
3. [and so on...]

**Expected behavior:** [What you expect to happen]

**Actual behavior:** [What actually happens]

**Reproduces how often:** [What percentage of the time does it reproduce?]

### Versions

SpotMenu version: [Version of SpotMenu]

You can get this information by highlighting SpotMenu.app (most likely in your Applications folder) in finder and pressing space.

MacOS version: [Version of MacOS]

You can get this information by clicking the  logo on the top left of your menu bar and selecting About This Mac.

### Additional Information

Any additional information, configuration or data that might be necessary to reproduce the issue.

**Comment by simon-wh** [comment]
To help out others who want to try it, I was able to compile it on my M1 Macbook Air following the instructions in the Readme.

The only additional changes I needed to do was to remove the following lines from the `Podfile` as they aren't compatible with Apple silicon, these just handle crash reporting and app updating so not very relevant to a local build anyway
```
pod 'Fabric'
pod 'Crashlytics'
pod 'Sparkle'
```

After that, I kept building and removing/commenting the lines it errored on (i.e. from missing package or item not found from the packages that were removed) until it successfully compiled. Effectively removing all the lines directly using stuff from the removed packages. After that, I was able to build and install it locally and now I've got a working native build.

**Comment by toineenzo** [comment]
@simon-wh awesome! I was looking if the fork was on your profile but couldn't find it. Could you perhaps share your build?

**Comment by jewellscott** [comment]
> @simon-wh awesome! I was looking if the fork was on your profile but couldn't find it. Could you perhaps share your build?

Seconding this request!

**Comment by MattijsE** [comment]
I've taken the liberty to add support for Apple Silicon (using the instructions provided by @simon-wh).

You can find the new .app over here https://github.com/MattijsE/SpotMenu/releases/tag/v1.10

**Comment by toineenzo** [comment]
> I've taken the liberty to add support for Apple Silicon (using the instructions provided by @simon-wh).
> 
> You can find the new .app over here https://github.com/MattijsE/SpotMenu/releases/tag/v1.10

Dank je wel, Thanks Mattijs for your build! Unfortunately I get the  'The application "..." can't be opened' error. I already tried to disable gatekeeper and give extra permission via terminal to the .app and executable file but didn't help. :(

**Comment by MattijsE** [comment]
> > I've taken the liberty to add support for Apple Silicon (using the instructions provided by @simon-wh).
> > You can find the new .app over here https://github.com/MattijsE/SpotMenu/releases/tag/v1.10
> 
> Dank je wel, Thanks Mattijs for your build! Unfortunately I get the 'The application "..." can't be opened' error. I already tried to disable gatekeeper and give extra permission via terminal to the .app and executable file but didn't help. :(

Yeah, you're right. I noticed the same when trying to run the .app on a different Mac.
Probably due to the fact that I didn't sign it with a paid developer account.

So you can always checkout my fork and make a .app yourself 😄

**Comment by Profitsarts** [comment]
> > > I've taken the liberty to add support for Apple Silicon (using the instructions provided by @simon-wh).
> > > You can find the new .app over here https://github.com/MattijsE/SpotMenu/releases/tag/v1.10
> > 
> > 
> > Dank je wel, Thanks Mattijs for your build! Unfortunately I get the 'The application "..." can't be opened' error. I already tried to disable gatekeeper and give extra permission via terminal to the .app and executable file but didn't help. :(
> 
> Yeah, you're right. I noticed the same when trying to run the .app on a different Mac. Probably due to the fact that I didn't sign it with a paid developer account.
> 
> So you can always checkout my fork and make a .app yourself 😄

Is that with Xcode? what may I search on google for a tutorial? thanks

**Comment by MattijsE** [comment]
> > > > I've taken the liberty to add support for Apple Silicon (using the instructions provided by @simon-wh).
> > > > You can find the new .app over here https://github.com/MattijsE/SpotMenu/releases/tag/v1.10
> > > 
> > > 
> > > Dank je wel, Thanks Mattijs for your build! Unfortunately I get the 'The application "..." can't be opened' error. I already tried to disable gatekeeper and give extra permission via terminal to the .app and executable file but didn't help. :(
> > 
> > 
> > Yeah, you're right. I noticed the same when trying to run the .app on a different Mac. Probably due to the fact that I didn't sign it with a paid developer account.
> > So you can always checkout my fork and make a .app yourself 😄
> 
> Is that with Xcode? what may I search on google for a tutorial? thanks

From the top of my head

- Open project
- Install pods
- Product --> archive
- Select archived app + click on 'Distribute app'
- Select the correct option (out of 4 I guess)
- You'll now have a folder with a .app inside
- Copy that to your Applications on your Mac

**Comment by Profitsarts** [comment]
> > > > > I've taken the liberty to add support for Apple Silicon (using the instructions provided by @simon-wh).
> > > > > You can find the new .app over here https://github.com/MattijsE/SpotMenu/releases/tag/v1.10
> > > > 
> > > > 
> > > > Dank je wel, Thanks Mattijs for your build! Unfortunately I get the 'The application "..." can't be opened' error. I already tried to disable gatekeeper and give extra permission via terminal to the .app and executable file but didn't help. :(
> > > 
> > > 
> > > Yeah, you're right. I noticed the same when trying to run the .app on a different Mac. Probably due to the fact that I didn't sign it with a paid developer account.
> > > So you can always checkout my fork and make a .app yourself 😄
> > 
> > 
> > Is that with Xcode? what may I search on google for a tutorial? thanks
> 
> From the top of my head
> 
> * Open project
> * Install pods
> * Product --> archive
> * Select archived app + click on 'Distribute app'
> * Select the correct option (out of 4 I guess)
> * You'll now have a folder with a .app inside
> * Copy that to your Applications on your Mac

Thanks a lot!

**Comment by Profitsarts** [comment]
> > > > > I've taken the liberty to add support for Apple Silicon (using the instructions provided by @simon-wh).
> > > > > You can find the new .app over here https://github.com/MattijsE/SpotMenu/releases/tag/v1.10
> > > > 
> > > > 
> > > > Dank je wel, Thanks Mattijs for your build! Unfortunately I get the 'The application "..." can't be opened' error. I already tried to disable gatekeeper and give extra permission via terminal to the .app and executable file but didn't help. :(
> > > 
> > > 
> > > Yeah, you're right. I noticed the same when trying to run the .app on a different Mac. Probably due to the fact that I didn't sign it with a paid developer account.
> > > So you can always checkout my fork and make a .app yourself 😄
> > 
> > 
> > Is that with Xcode? what may I search on google for a tutorial? thanks
> 
> From the top of my head
> 
> * Open project
> * Install pods
> * Product --> archive
> * Select archived app + click on 'Distribute app'
> * Select the correct option (out of 4 I guess)
> * You'll now have a folder with a .app inside
> * Copy that to your Applications on your Mac

Hi!
Sorry but I have no experience with Xcode.

I used Xcode 13.2.1 and the only result I get is a file called "MusicPlayer [date].xcarchive"

When you say "Install pods" I interpreted selecting the "pods" folder and hit the play button, then in the "Select archived app + click on 'Distribute app' just have to options and the only one that generates a file is the second one 'Archive: export a copy of this file' (1st is 'Built products')

I guess it's easier than this but, no clue :(

Thanks!

**Comment by MattijsE** [comment]
> 

To install the pods, you'll have to navigate in a terminal to the root of the project (where Podfile & Podfile.lock are) and execute `pod install`.

**Comment by MattijsE** [comment]
> 
@Profitsarts 
To install the pods, you'll have to navigate in a terminal to the root of the project (where Podfile & Podfile.lock are) and execute `pod install`.

**Comment by Profitsarts** [comment]
> @Profitsarts
> To install the pods, you'll have to navigate in a terminal to the root of the project (where Podfile & Podfile.lock are) and execute `pod install`.

Thanks for the helping hand :)
But, after install 'sudo gem install cocoapods' and run 'pod install' I get this: 

[NOTE]
You may have encountered a bug in the Ruby interpreter or extension libraries.
Bug reports are welcome.
For details: https://www.ruby-lang.org/bugreport.html

[IMPORTANT]
Don't forget to include the Crash Report log file under
DiagnosticReports directory in bug reports.

zsh: abort      pod install

-----------

Then When I go to distribute, I just have two options: 

1. 'Archive: export a copy of this file'
2.  'Built products'

If a chose the 2nd I get a folder with a '.xcarchive' file on it, just that :(


Sorry, I am so limited doing this things

**Comment by kmikiy** [comment]
closing because I'm rewriting the app from scratch https://github.com/kmikiy/SpotMenu/issues/279#issuecomment-2987565341
